drivers/input/i8042prt/pnp.c

```diff
@@ -470,6 +516,8 @@ StartProcedure(
         DeviceExtension->Flags & MOUSE_STARTED &&
         !(DeviceExtension->Flags & MOUSE_INITIALIZED))
     {
+        /* Detect the mouse type here, rather than in i8042DetectMouse, fixes lockup on BootCD */
+        i8042DetectMouseType(DeviceExtension);
```

Comment:
As already mentioned in the original PR by @BenNottelling , the mouse type is also set in the `i8042MouResetIsr()` function, see https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/input/i8042prt/mouse.c;hb=6d238e451345dfd4f5b8178d4d515801546fe95d#l629
Hence I have a question: When is this `i8042MouResetIsr()` function called, relative to this new `i8042DetectMouseType()` of yours? Before? After?
And would it be then possible to not duplicate the functionality of setting the MouseType?

Reply:
`i8042DetectMouseType()` is called right before the interrupt for the mouse is connected. The mouse ISR is the one calling `i8042MouResetIsr`, hence before.

Reply:
> As already mentioned in the original PR by @BenNottelling , the mouse type is also set in the `i8042MouResetIsr()` function, see https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/input/i8042prt/mouse.c;hb=6d238e451345dfd4f5b8178d4d515801546fe95d#l629
> Hence I have a question: When is this `i8042MouResetIsr()` function called, relative to this new `i8042DetectMouseType()` of yours? Before? After?
> And would it be then possible to not duplicate the functionality of setting the MouseType?

I would be willing to explore the possibility of not having to duplicate. What I'm given to understand is that the second reset does that. This causes the trackpad on some dell laptops to not work. This is where hwhacks was introduced. So if  someone is willing to give me debug logs for the D latitude series Dell laptops, that would be great. I can try to get the trackpad to work even with the second reset.

Or we can simply make a quick fix like it is happening currently.